syntax = "proto3";

package plagiarism_detection;

service PlagiarismDetectionService {
    rpc Ping(PingRequest) returns (PingResponse) {}
    rpc Login(LoginRequest) returns (LoginResponse) {}
    rpc Register(RegisterRequest) returns (RegisterResponse) {}
    rpc UploadFile(UploadFileRequest) returns (UploadFileResponse) {}
    rpc CheckTwoFiles(CheckTwoFilesRequest) returns (CheckTwoFilesResponse) {}
}

message PingRequest {
}

message PingResponse {
    int32 status = 1;
}

message LoginRequest {
    string username = 1;
    string password = 2;
}

message LoginResponse {
    int32 status = 1;
    string token = 2;
}

message RegisterRequest {
    string username = 1;
    string password = 2;
}

message RegisterResponse {
    int32 status = 1;
}

message UploadFileRequest {
    string token = 1;
    string file = 2;
}

message UploadFileResponse {
    string token = 1;
    string file_id = 2;
}

message CheckTwoFilesRequest {
    string token = 1;
    string file_id_1 = 2;
    string file_id_2 = 3;
}

message CheckTwoFilesResponse {
    repeated int32 line_numbers = 1;
}